crates/metal-llvm/src/stmt.rs

```diff
@@ -47,14 +49,67 @@ impl CodeGenValue for FunctionDefinition {
     }
 }
 
+impl CodeGenValue for Constant {
+    fn codegen_value(
+        &self,
+        llvm: &mut crate::LLVMRefs,
+        module: &metal_mir::parcel::Module,
+    ) -> LLVMValueRef {
+        let cname = CString::new(self.name).unwrap();
+
+        unsafe {
+            let global_var = LLVMAddGlobal(
+                llvm.module,
+                self.ty.codegen_type(llvm, module),
+                cname.as_ptr(),
+            );
+
+            match self.expr {
+                Expr::Literal(_) => {
+                    let val = self.expr.codegen_value(llvm, module);
+                    LLVMSetInitializer(global_var, val);
```

Comment:
What does this do? LLVM docs on this function [don't go into much detail](https://llvm.org/doxygen/classllvm_1_1GlobalVariable.html#a095f8f031d99ce3c0b25478713293dea).

Reply:
> What does this do? LLVM docs on this function [don't go into much detail](https://llvm.org/doxygen/classllvm_1_1GlobalVariable.html#a095f8f031d99ce3c0b25478713293dea).

Essentially just sets the initial (and in this case permanent) value of the constant variable